AMRC-FactoryPlus/cs-serviceclient
# Added .NET 4.8.1 as a build target and refactored to C#8 to accommodate

On of the major use cases for this library is for Unity development. We overlooked the actual capabilities of the Unity platform, as it doesn't _yet_ support .NET 6. The library now targets .NET 4.8.1, and the output DLL seems to work in Unity.

To support this, the code needed refactoring since it utilised a lot of C#9 and 10 features that aren't supported in .NET 4.8.1.

#### How to test:

1. Might need the .NET 4.8.1 SDK: https://dotnet.microsoft.com/en-us/download/dotnet-framework/net481
2. Open the library in Rider
3. Clean the solution and then rebuild
4. Notice the /bin/Debug folder now has 4.8.1 and 6.0
5. Run the sample app and notice that it still functions
